brave/brave-core
# fix(privacy): Use HttpsUpgradeService / HttpsOnlyModeAllowlist on iOS

- Update iOS to use `HttpsOnlyModeAllowlist` via `BraveHttpsUpgradeService ` (wrapper for `HttpsUpgradeService`).
    - Additionally, by using `HttpsOnlyModeAllowlist` we will get time-based expiration of allowlist hosts.
- Update iOS to clear allow list when HTTPS by Default preference is changed, or browser data is cleared.
- Additionally decouples iOS `HTTPSUpgradeExceptionsService` from user allow list.

Resolves https://github.com/brave/brave-browser/issues/40710

## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. In Shields settings, set 'Upgrade Connections to HTTPS' to 'Standard'.
2. Visit `http://badssl.com`. Verify url is upgraded to `https://badssl.com`
    - Verifies auto-upgrade to HTTPS.
3. Scroll down and tap/visit any of the 'HTTP' section links. Verify the HTTP pages load as expected.
    - Verifies fallback to http when https upgrade fails.
4. Make note of at least one of the 'HTTP' section links tapped / visited, for example visit `http://http.badssl.com/`.
5. In Shields settings, set 'Upgrade Connections to HTTPS' to 'Strict'.
6. Visit the same link from step 4. Verify you are presented with the HTTPS strict-mode interstitial.
    - Verifies the allow list is reset when preference changes
7. On the HTTPS strict-mode interstitial, tap 'Proceed'. Verify website loads as expected.
8. Copy the url for the tab, open a new tab and visit the same url. Verify the interstitial is not shown.
    - Verifies the allow list is used for strict mode
9. Open Shields settings, scroll to Clear Private Data section, toggle on 'Browsing History' and tap 'Clear Data Now'.
10. Visit the url from step 8. Verify the HTTPS strict-mode interstitial is shown again.
    - Verifies the allow list is cleared when browsing history is cleared
